src/sentry/deletions/base.py

```diff
@@ -187,9 +199,9 @@ def chunk(self, num_shards=None, shard_id=None):
             if not queryset:
                 return False
 
-            self.delete_bulk(queryset)
+            has_more = self.delete_bulk(queryset)
             remaining -= query_limit
-        return True
```

Comment:
Always returning true results in additional tasks being spawned for every delete we do. These tasks do no work but still consume resources. Now only if the last `delete_bulk` operation has more work will we spawn additional tasks.